cms/djangoapps/contentstore/views/course.py

```diff
@@ -989,101 +984,37 @@ def grading_handler(request, course_key_string, grader_index=None):
                 return JsonResponse()
 
 
-# pylint: disable=invalid-name
-def _add_tab(request, tab_type, course_module):
+def _refresh_course_tabs(request, course_module):
     """
-    Adds tab to the course.
+    Automatically adds/removes tabs if changes to the course require them.
     """
-    # Add tab to the course if needed
-    changed, new_tabs = add_extra_panel_tab(tab_type, course_module)
-    # If a tab has been added to the course, then send the
-    # metadata along to CourseMetadata.update_from_json
-    if changed:
-        course_module.tabs = new_tabs
-        request.json.update({'tabs': {'value': new_tabs}})
-        # Indicate that tabs should not be filtered out of
-        # the metadata
-        return True
-    return False
 
+    def update_tab(tabs, tab_type, tab_enabled):
+        """
+        Adds or removes a course tab based upon whether it is enabled.
+        """
+        tab_panel = {
+            "type": tab_type.name,
+            "name": tab_type.title,
+        }
+        has_tab = tab_panel in tabs
+        if tab_enabled:
+            if not has_tab:
+                tabs.append(CourseTab.from_json(tab_panel))
+        elif has_tab:
+            tabs.remove(tab_panel)
```

Comment:
Nit: perhaps

    if tab_enabled and not has_tab:
        ...
    elif not tab_enabled and has_tab:
        ...